ui/src/app/dashboard/dashboard.component.spec.ts

```diff
@@ -67,9 +70,55 @@ describe('DashboardComponent', () => {
     expect(testComponent).toBeTruthy();
   }));
 
+  it("should create expected amount of cards", fakeAsync(() => {
+    let matCards = fixture.debugElement.queryAll(By.css('mat-card'));
+    expect(matCards.length).toEqual(CARD_NUM);
+  }));
+
+  it('should navigate to job-list page (fake) when status counts are clicked', fakeAsync(() => {
+    let countAnchor = fixture.debugElement.query(By.css("jm-dashboard a")).nativeElement;
+    countAnchor.click();
+    fixture.detectChanges();
+    tick();
+    expect(fixture.debugElement.query(By.css("div")).nativeElement.textContent).toEqual("fake job-list page");
+  }));
+
+  it('should has status as url param when totalSummaryComponent links are clicked', fakeAsync(() => {
+    let totalSummaryAnchor = fixture.debugElement.query(By.css("jm-total-summary .count a")).nativeElement;
+    let status = fixture.debugElement.query(By.css("jm-total-summary .status")).nativeElement.textContent.trim();
+
+    totalSummaryAnchor.click();
+    fixture.detectChanges();
+    tick();
+    testJobListComponent = fixture.debugElement.query(By.css("jm-test-job-list-component")).componentInstance;
+    expect(testJobListComponent.route.snapshot.queryParams['q'].indexOf(status)).toBeGreaterThan(-1);
+  }));
+
+  it("should has status and label as url params when groupedSummaryComponent links are clicked", fakeAsync(() => {
+    let groupedSummaryAnchor = fixture.debugElement.query(By.css("jm-grouped-summary tr td.count a")).nativeElement;
+    groupedSummaryAnchor.click();
+    tick();
+    testJobListComponent = fixture.debugElement.query(By.css("jm-test-job-list-component")).componentInstance;
+    let q = testJobListComponent.route.snapshot.queryParams['q'];
+    // this test is based on the hard-coded aggregation response
+    expect(q.indexOf(TEST_AGGREGATION_RESPONSE.aggregations[0].key)).toBeGreaterThan(-1);
+    expect(q.indexOf(TEST_AGGREGATION_RESPONSE.aggregations[0].entries[0].statusCounts.counts[0].status)).toBeGreaterThan(-1);
+  }));
+
   @Component({
     selector: 'jm-test-app',
     template: '<router-outlet></router-outlet>'
   })
   class AppComponent {}
+
+  @Component({
+    selector: 'jm-test-job-list-component',
+    template: '<div>fake job-list page</div>'
+  })
+  class TestJobListComponent {
+    public route: ActivatedRoute;
+      constructor(activatedRoute : ActivatedRoute) {
```

Comment:
In Typescript, there's syntactic sugar for this pattern, so you shouldn't need to make these Java-style constructors in general. The following achieves the same result as what you have currently:

```
class TestJobListComponent {
  constructor(public activatedRoute: ActivatedRoute) {}
}
```

Reply:
Adapted, thanks!